.github/workflows/regression_yamls/llama2-finetune.yaml

```diff
@@ -0,0 +1,121 @@
+integrations:
+- integration_type: git_repo
+  git_repo: mosaicml/llm-foundry
+  git_branch: main
+  pip_install: -e .[gpu]
+
+command: |
+  cd llm-foundry/scripts
+  composer train/train.py /mnt/config/parameters.yaml
+  python inference/convert_composer_to_hf.py \
+    --composer_path ./llama2-finetune-regression/checkpoints/latest-rank0.pt \
+    --hf_output_path ./hf-checkpoints/
+image: mosaicml/llm-foundry:1.13.1_cu117-latest
+name: llama2-finetune-regression
+
+compute:
+  gpus: 8
+
+parameters:
+  tokenizer_name: meta-llama/Llama-2-13b-hf
+  max_seq_len: 4096
+  global_seed: 17
+
+  run_name: llama2-finetune-regression
+  max_split_size_mb: 512
+
+  model:
+    name: hf_causal_lm
+    init_device: mixed
```

Comment:
q: what is init_device and why does it vary across these yamls?

Reply:
`init_device` controls what device the model is created on before being passed to composer to get FSDP wrapped. The options are:
- `cpu`: init model on cpu on all ranks. 8 copies of model in CPU RAM before being passed to composer
- `meta`: init model as [meta tensors](https://pytorch.org/torchdistx/latest/fake_tensor_and_deferred_init.html). this is very fast, and then allows init to happen directly on GPU during FSDP wrapping (or whenever it is materialized).
- `mixed`: init on cpu on local rank 0, and meta on all other ranks. this allows us to load weights from the huggingface hub and then FSDP wrap without creating 8 copies of the model in CPU RAM. The loaded weights are synced between all ranks during FSDP wrapping.